Network/HTTP/Semantics/Types.hs

```diff
@@ -76,9 +77,23 @@ data OutBodyIface = OutBodyIface
     , outBodyPushFinal :: Builder -> IO ()
     -- ^ Push the final chunk
     --
-    -- Using this function instead of 'outBodyPush' can be used to guarantee that the final
-    -- HTTP2 DATA frame is marked end-of-stream; with 'outBodyPush' it may happen that
-    -- an additional empty DATA frame is used for this purpose.
+    -- Using this function instead of 'outBodyPush' can be used to guarantee
+    -- that the final HTTP2 DATA frame is marked end-of-stream; with
+    -- 'outBodyPush' it may happen that an additional empty DATA frame is used
+    -- for this purpose.
+    --
+    -- /Postcondition/: After calling 'outBodyPushFinal', 'outBodyCancel' will
+    -- be a no-op.
```

Comment:
I think it would be good to combine this comment with the previous: there are _two_ differences between `outBodyPushFinal` and `outBodyPush`; the empty data frame, and the postcondition.